apps/agora/app/src/app/app.component.ts

```diff
@@ -9,6 +11,36 @@ import { FooterComponent } from '@sagebionetworks/agora/ui';
   templateUrl: './app.component.html',
   styleUrl: './app.component.scss',
 })
-export class AppComponent {
-  title = 'agora-app';
+export class AppComponent implements OnInit {
+  router = inject(Router);
+  activatedRoute = inject(ActivatedRoute);
+  titleService = inject(Title);
+  metaService = inject(Meta);
+
+  ngOnInit(): void {
+    this.router.events.pipe(filter((event) => event instanceof NavigationEnd)).subscribe(() => {
+      const route = this.getChildRoute(this.activatedRoute);
+      route.data.subscribe((data: any) => {
+        if (data.title) {
+          this.titleService.setTitle(data.title);
+        } else {
+          this.titleService.setTitle('Agora');
+        }
```

Comment:
Replace with a more concise implementation.

```suggestion
        this.titleService.setTitle(data.title || 'Agora');
```

Reply:
nice!